README.md

```diff
@@ -101,45 +99,140 @@ The back-end service implements the following protocol:
+  - `canHide`: boolean, whether the column can be hidden by the user
+    (tip: to include columns that are always hidden, put them only in the data object, not in the columns config)
+- `filters`: array, the possible filters configuration of the table
+  - `id`: string, unique id for the filter
+  - `type`: string, how to present the filter: `select`, `input` (text), `datepicker`
+  - `display`: string, how to show the filter in the UI
+    - `generic`: a google-like search textbox to search across multiple fields, always shown
+    - `visible`: field-specific filter control, always shown
+    - `optional`: field-specific filter control, hidden behind 'Extra filters' button 
+  - `label`: string, label to show next to filter control
+  - `field`: string, key of the field (column) to filter on
+  - `fields`: string[], keys of the fields to filter on, for filter type `generic`
+  - `operator`: string, operator to compare the value with, `=` / `ILIKE`
+  - `options`: {id: string, label: string}[], content of dropdown list for filter type `select`
+  - `placeholder`: string
+  - `value`: string | {id: string, label: string}[]
+- `baseFilters`: array, a set of fixed and hidden filters to always apply to data requests
+  - `fields`: string[], keys of the fields to filter on
+  - `operator`: string, operator to compare the value with, `=` / `ILIKE`
+  - `value`: string | {id: string, label: string}[]
+- `options`: object, other configuration for the table
+  - `defaultSortOrder`: { key: string, order: 'asc'|'desc' }, initial sort order
+  - `loadDataMessage`: string, shown while loading
+  - `noDataMessage`: string, shown when empty
+  - `errorMessage`: string, shown when loading failed
+  - `pageSize`: number, initial page size (in rows)
+  - `pageSizeOptions`: number[], choices presented for page size in the UI
+  - `resetSortOrderOnFilter`: boolean, return to default sort order when changing filters
+  - `columnDateTimeFormat`: string, default format for date/time columns, see https://angular.io/api/common/DatePipe
+  - `columnDateFormat`: string, default format for date columns, see https://angular.io/api/common/DatePipe
+  - `storageIdentifier`: string, unique key for persisting column state
+  - `persistTableConfig`: boolean, true to persist column state
+  - `translations`: object, translation strings to replace default (dutch) text in UI (see `smart-table.types.ts`)
+
+### Custom Columns
+
+The default column types are text, number, date and dateTime. Sometimes a special column type is needed.
+
+In the showcase example there is a custom column for the star rating.
+
+To implement a custom column:
+- Implement a component that subclasses `Cell` from ngx-table. Use `rating.component.ts` as an example.
+- Include this custom column component in the smart table attributes as a custom column type:
+  ```html
+  <aui-smart-table
+      [columnTypes]="myCustomColumns"
+  ```
+  ```js
```

Comment:
JSON too?

Reply:
fails to format well because the keys need to be quoted and the type information causes issues (not valid json).